rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -218,6 +224,13 @@ fn wireless_config_to_dbus(conn: &WirelessConnection) -> NestedHash {
     let wireless: HashMap<&str, zvariant::Value> = HashMap::from([
         ("mode", Value::new(config.mode.to_string())),
         ("ssid", Value::new(config.ssid.to_vec())),
+        (
+            "assigned-mac-address",
+            Value::new(match &conn.base.mac_address {
+                Some(mac) => mac.to_string(),
+                None => "".to_string(),
```

Comment:
because of this, maybe we could place `mac_address_string()` into `BaseConnection`